src/utils.jl

```diff
@@ -35,20 +30,97 @@ function render(s::LaTeXString; debug=false, name=tempname(), command="\\Large")
         write(f, doc)
     end
-    cd(dirname(name)) do 
-        cmd = `lualatex --interaction=batchmode $(name).tex`
-        debug || (cmd = pipeline(cmd, devnull))
-        run(cmd)
-    end
+
+    return nothing
+end
+
+
+function _openfile(name; ext="pdf")
     if Sys.iswindows()
-        run(`cmd /c "start $(name).pdf"`, wait=false)
+        run(`cmd /c "start $(name).$(ext)"`, wait=false)
     elseif Sys.islinux()
-        run(`xdg-open $(name).pdf`, wait=false)
+        run(`xdg-open $(name).$(ext)`, wait=false)
     elseif Sys.isapple()
-        run(`open $(name).pdf`, wait=false)
+        run(`open $(name).$(ext)`, wait=false)
     elseif Sys.isbsd()
-        run(`xdg-open $(name).pdf`, wait=false)
+        run(`xdg-open $(name).$(ext)`, wait=false)
+    end
+
+    return nothing
+end
+
+
+"""
+    render(::LaTeXString[, ::MIME"mime"]; debug=false, name=tempname(), command="\\Large", open=true)
+
+Display a standalone document with the given input. Supported MIME-type strings are
+"application/pdf" (default), "application/x-dvi", "image/png" and "image/svg+xml".
+"""
+function render(s::LaTeXString; debug=false, name=tempname(), command="\\Large", open=true)
+    return render(s, MIME("application/pdf"); debug=debug, name=name, command=command, open=open)
+end
+
+
+function render(s::LaTeXString, ::MIME"application/pdf"; debug=false, name=tempname(), command="\\Large", open=true)
+    _writetex(s; name=name, command=command)
+
+    cd(dirname(name)) do
+        cmd = `lualatex --interaction=batchmode $(name).tex`
+        debug || (cmd = pipeline(cmd, devnull))
+        run(cmd)
+    end
+
+    if open
+        _openfile(name; ext="pdf")
     end
+
+    return nothing
+end
+
+
+function render(s::LaTeXString, ::MIME"application/x-dvi"; debug=false, name=tempname(), command="\\Large", open=true)
+    _writetex(s; name=name, command=command)
+
+    cd(dirname(name)) do
+        cmd = `dvilualatex --interaction=batchmode $(name).tex`
+        debug || (cmd = pipeline(cmd, devnull))
+        run(cmd)
+    end
+
+    if open
+        _openfile(name; ext="dvi")
+    end
+
+    return nothing
+end
+
+
+function render(s::LaTeXString, ::MIME"image/png"; debug=false, name=tempname(), command="\\Large", open=true, dpi=300)
+    render(s, MIME("application/x-dvi"); debug=debug, name=name, command=command, open=false)
+
+    cmd = `dvipng -bg Transparent -D $(dpi) -T tight -o $(name).png $(name).dvi`
+    debug || (cmd = pipeline(cmd, devnull))
+    run(cmd)
+
+    if open
+        _openfile(name; ext="png")
+    end
+
+    return nothing
+end
+
+
+function render(s::LaTeXString, ::MIME"image/svg+xml"; debug=false, name=tempname(), command="\\Large", open=true)
+    render(s, MIME("application/x-dvi"); debug=debug, name=name, command=command, open=false)
+
+    cmd = `dvisvgm -n -v 0 -o $(name).svg $(name).dvi`
+    debug || (cmd = pipeline(cmd, devnull))
+    run(cmd)
+
+    if open
+        _openfile(name; ext="svg")
+    end
+
```

Comment:
Adding 
```julia
    if callshow && displayable(MIME("image/svg+xml"))
        Base.open(fname * ".svg") do f
            display(MIME("image/svg+xml"), read(f, String))
        end
    end
```

and a corresponding `callshow` kwarg to the function would allow for displaying the SVG in the user's environment (for example the plot pane in vscode). 

I know that this would work nicely with VSCode and SVG, but I'm guessing that it'll work for some other mime-types as well.  

This can probably be done with a nicer interface though. I'm thinking that it could be nice to display the file in the user's environment by default (if possible) and to fallback to opening the file externally.

Reply:
That's nice to defer to `displayable` and support all mime-types this way. I'll default to `display`ing if possible and opening if not, in that order of priority, but leaving `callshow=true, open=true` as keywords if the user wants more control.